### PR TITLE
feat: adds onRequest hook, which passes request from Wreck.request

### PR DESCRIPTION
**Pull request by AVVS**
BREAKING CHANGE: upgrades Hapi to 15.x.x

### CODE REVIEW

**Comment by osukaa** [comment]
I'm cc/ @nathanmarks because he wanted something similar to this. Perhaps you can add what he wanted in the same PR

File: package.json [hunk 1]
```diff
@@ -15,16 +15,19 @@
...
-    "wreck": "9.X.X"
+    "boom": "4.x.x",
+    "hoek": "4.x.x",
+    "joi": "9.x.x",
+    "wreck": "10.x.x"
+  },
+  "peerDependencies": {
+    "hapi": "14.x.x || 15.x.x"
   },
   "devDependencies": {
```
Comment (AVVS) [inline package.json]: could you explain more? I'm not very knowledgeable about the code-style here. What exactly should I do with the `.x.x` ?

Comment (AVVS) [inline package.json]: nvm, you were talking about dev deps, will change that

**Comment by osukaa** [comment]
Not really, I just need the branch to be updated and resolved the merge conflicts :)

**Comment by spanditcaa** [comment]
@AVVS @rexxars care to update and resolve the merge conflicts ?

**Comment by spanditcaa** [comment]
addressed merge conflicts.

**Comment by spanditcaa** [comment]
failing on node 9.2 only - looking into it...

**Comment by spanditcaa** [comment]
@osukaa @Marsup @hueniverse I wanted to merge the outstanding pre-v17 features before the hapi17 branch so we'd have that functionality for <17 users -- but I'm seeing a travis failure here that I don't understand.  All the tests appear to pass and coverage is at 100%, but travis is marking npm test as failed on node 9.2 only.  Do you have any insight as to what might be causing this?

**Comment by Marsup** [comment]
The fix for https://github.com/hapijs/lab/issues/748 would have to be backported to 14 for that.

**Comment by Marsup** [comment]
Monitor https://github.com/hapijs/lab/pull/792.

**Comment by Marsup** [comment]
@spanditcaa it's now green.

**Pull request merged**